From 0369afc7be852b49c8641280c7adbd6b52694696 Mon Sep 17 00:00:00 2001
From: Alessio Ferri <alessio.ferri.3012@gmail.com>
Date: Wed, 17 Jun 2026 20:55:58 +0000
Subject: [PATCH 3/8] bcma: pcie2: request HT on BCM4352/BCM4360

Set HAVEHTREQ in the PCIe2 core's clock control on chip rev 3, as
Broadcom's driver does.
---
 drivers/bcma/driver_pcie2.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/bcma/driver_pcie2.c b/drivers/bcma/driver_pcie2.c
index cf889fc62..fd8fab982 100644
--- a/drivers/bcma/driver_pcie2.c
+++ b/drivers/bcma/driver_pcie2.c
@@ -170,6 +170,9 @@ void bcma_core_pcie2_init(struct bcma_drv_pcie2 *pcie2)
 	case BCMA_CHIP_ID_BCM4360:
 	case BCMA_CHIP_ID_BCM4352:
 		pcie2->reqsize = 1024;
+		if (ci->rev == 3)
+			pcie2_set32(pcie2, BCMA_CLKCTLST,
+				    BCMA_CLKCTLST_HAVEHTREQ);
 		break;
 	default:
 		pcie2->reqsize = 128;
